src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -152,6 +152,27 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           resolver.resolve(fileRelativeUrl`%><><%=`, packageRoot), undefined);
     });
+
+    test('resolves a URL with no pathname', () => {
+      const foo = rootedFileUrl`1/2/foo.html?baz#bat`;
+      const bar = rootedFileUrl`1/2/bar.html`;
```

Comment:
Could you split this test up? There's a combinatorial set of conditions of {path, fragment, query} * {base, target}

Reply:
I removed the vars and ordered the items progressively.  Maybe is okay now?